### DISCUSSION TITLE
app routing not working

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#49768] [App Router]
my folder is like
app > page.jsx
app > cookies > page.jsx

when i do a dev build (npm run dev), the localhost:3000/cookies works fine, but when i do a build, and host it, the /cookies return a 404. im using firebase hosting and the url is the default firebase hosting
(project id).web.app

**Comment by ghost** [comment]
dev build localhost:3000/cookies works but
(project id).web.app/cookies returns 404

**Comment by icyJoseph** [comment]
Are you using `export` mode? https://nextjs.org/docs/pages/building-your-application/deploying/static-exports

Or what happens if you go to `(project id).web.app/cookies/`?

**Comment by ghost** [comment]
I'm using `output: 'export',`,
but the issue was with how I was configuring my firebase hosting.
I fixed it by adding:
"rewrites": [
      {
        "source": "/cookies",
        "destination": "/cookies.html"
      },
      {
        "source": "/impressum",
        "destination": "/impressum.html"
      },
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]

**Reply by icyJoseph** [reply]
Yeah exactly why I asked about export, great that you went ahead and figured it out! There's a thread out there where people share lots of server configs too.

**Reply by ghost** [reply]
Where can I find that thread? Thanks

**Reply by icyJoseph** [reply]
https://github.com/vercel/next.js/discussions/10522

You'll see there that some also use trailing slash as escape hatch, for example here's hosting services that do not allow modification to how the static server works, in that situation trailing slashes help
